lib/compatibility.bzl

```diff
@@ -0,0 +1,64 @@
+"""TODO(trybka): Write module docstring."""
+
+load(":selects.bzl", "selects")
+
+INCOMPATIBLE_TARGET="@platforms//:incompatible"
+INCOMPATIBLE_SETTING="@platforms//:incompatible_setting"
+
+def _none_of_without_no_match(settings):
+    settings_name = "_or_".join([s.split(":")[1] for s in settings])
+    name = settings_name.replace(":", "-").replace("//", "").replace("/", "_")
+    compat_name = "incompatible_with_" + name
+    if not native.existing_rule(compat_name):
+        # We need to make (or re-use?) constraint values within a single instance to avoid a duplicate label.
+        native.constraint_value(name = compat_name, constraint_setting = INCOMPATIBLE_SETTING)
+    return selects.with_or({"//conditions:default": [], tuple(settings): [":" + compat_name]})
+
+def _any_of_without_no_match(settings):
+    settings_name = "_or_".join([s.split(":")[1] for s in settings])
+    name = settings_name.replace(":", "-").replace("//", "").replace("/", "_")
+    compat_name = "compatible_with_any_of_" + name
+    if not native.existing_rule(compat_name):
+        native.constraint_value(name = compat_name, constraint_setting = INCOMPATIBLE_SETTING)
+    return selects.with_or({tuple(settings): [], "//conditions:default": [":" + compat_name]})
+
+def _all_of_without_no_match(settings):
+    settings_name = "_and_".join([s.split(":")[1] for s in settings])
+    name = settings_name.replace(":", "-").replace("//", "").replace("/", "_")
+    if not native.existing_rule(name):
+        selects.config_setting_group(name = name, match_all = settings)
+    compat_name = "compatible_with_all_of_" + name
```

Comment:
Neat idea!

Reply:
What's the difference between this new value and just `@platforms//:incompatible`?

Reply:
`@platforms//:incompatible` cannot appear twice in this attribute--so if two selects resolve to incompatible, there will be an error.